Gameplay.js

```diff
@@ -53,13 +53,31 @@ class Gameplay {
 		this.board0.render(document.getElementById("board0"), this, true, true);
 		this.board1.render(document.getElementById("board1"), this, true, true);
 	}
-	
-	
-	clickSpace(cell,blocked) {
-		if (!blocked && !cell.isHit) {
-			cell.isHit = true;
-			this.renderBoards(true);	
-			this.switchTurns(true);
+
+
+	clickSpace(cell, isCurrentPlayer) {
+		if (this.isSetup)
+		{
+			if (!isCurrentPlayer && !cell.isHit) {
+				cell.isHit = true;
+				if (cell.hasShip)
+				{
+					if (!this.turn){
```

Comment:
There's a lot of duplicate code in this if/else. It could be better to assign which board to use to a variable instead.